build.xml

```diff
@@ -8,119 +8,268 @@
+    </xslt>
+    
+    <echo message="   STEP 2: Expanding abstracts"/>
+    <xslt in="${xspec.dir}/${xspec.base}_dsdl_included.sch" 
+      out="${xspec.dir}/${xspec.base}.sch" 
+      style="${xspec.project.dir}/src/schematron/iso-schematron/iso_abstract_expand.xsl" 
+      force="true"
+      >
+      <factory name="net.sf.saxon.TransformerFactoryImpl"/>
+    </xslt>
+    
+    <echo message="   STEP 3: Convert Schematron to XSL"/>    
+    <xslt 
+      in="${xspec.dir}/${xspec.base}.sch"
+      out="${xspec.compiled.xsl}" 
+      style="${xspec.project.dir}/src/schematron/iso-schematron/iso_svrl_for_xslt2.xsl" 
+      force="true"
+      >
+      <factory name="net.sf.saxon.TransformerFactoryImpl"/>
+      <param name="allow-foreign" expression="true"/>
+      <param name="phase" expression="${xspec.phase}"/>
+    </xslt>
+    
+    <echo message="   STEP 4: Convert XSpec (into a format that references the XSL generated from the Schematron)"/>    
+    <xslt 
+      in="${xspec.xml}"
+      out="${xspec.compiled.xml}" 
+      style="${xspec.project.dir}/src/schematron/schut-to-xspec.xsl" 
+      force="true"
+      >
+      <factory name="net.sf.saxon.TransformerFactoryImpl"/>
+      <param name="stylesheet" expression="${xspec.compiled.xsl.param}"/>
+    </xslt>
+
+    <echo message="COMPILATION COMPLETE"/>
+    
+   </target>
+  
+   
 
+  
   <target name="xspec"
-    description="Runs the XSpec unit tests in ${xspec.xml}">
-    <xslt in="${xspec.xml}" 
+    description="Runs the XSpec unit tests in ${xspec.compiled.xml}" depends="copyXSpec,compileSchematron">
+    
+    <echo message="Creating Test Stylesheet..."/>
+    <xslt in="${xspec.compiled.xml}" 
       out="${xspec.xsl}"
       style="${xspec.project.dir}/src/compiler/generate-xspec-tests.xsl"
-      classpath="${saxon.jar}">
-			<factory name="net.sf.saxon.TransformerFactoryImpl" />
-		</xslt>
-		<!-- Can't specify initial template with <xslt> task so have to
-		     fall back to running Saxon as Java app. -->
-		<java classname="net.sf.saxon.Transform"
-					classpath="${saxon.jar}"
-					fork="true">
-			<arg value="-ext:on" />
-			<arg value="-s:${xspec.xml}" />
-			<arg value="-o:${xspec.result.xml}" />
-			<arg value="-xsl:${xspec.xsl}" />
-			<arg value="-it:{http://www.jenitennison.com/xslt/xspec}main" />
-		</java>
+      force="true"
+      >     
+      <factory name="net.sf.saxon.TransformerFactoryImpl" />     
+    </xslt>
+    
+    <echo message="Running Tests..."/>
+    
+    <xslt in="${xspec.compiled.xml}" 
+      out="${xspec.result.xml}"
+      style="${xspec.xsl}"
+      force="true">
+      <factory name="net.sf.saxon.TransformerFactoryImpl" >
+        <attribute 
+          name="http://saxon.sf.net/feature/initialTemplate" 
+          value="{http://www.jenitennison.com/xslt/xspec}main"/>
+        <attribute
+          name="http://saxon.sf.net/feature/allow-external-functions" 
+          value="true"/>
+      </factory>
+    </xslt>
```

Comment:
`saxon.custom.options` is defined before, but not used here. Why? (Is it possible in the first place?)

Reply:
I guess `saxon.custom.options` is used by oXygen. I can see it is defined with this comment:

```xml
  <!-- OXYGEN PATCH needed to create a custom Saxon, for example  -warnings:recover -strip:none -opt:10 -dtd:off -l:off -versionmsg:off -expand:on -outval:fatal -val:lax -->
  <property name="saxon.custom.options" value=""/>
  <!-- OXYGEN PATCH END -->
```

I wonder if it is is worth removing oXygen customizations that are currently not needed for this feature and work with oXygen for a smooth integration of XSpec. This should also address https://github.com/xspec/xspec/pull/138#issuecomment-360111771 raised by @markdunnoup.

Shall we just remove the `saxon.custom.options` if not used? Do you see any side effect?

Reply:
We may just remove it, but...
This PR changes `<java>` to `<xslt>`. Is it easy to implement the feature of `saxon.custom.options` in `<xslt>`? I'm afraid not.

Reply:
I reintroduced the use of `<java>` instead of `<xslt>` in https://github.com/xspec/xspec/commit/3bcb2cd3ecaa6ff117e3193546673941544eff4f#diff-2cccd7bf48b7a9cc113ff564acd802a8R227. I have kept `saxon.custom.options` as a result.